mava/systems/jax/ippo/networks.py

```diff
@@ -196,27 +201,51 @@ def make_discrete_networks(
         key: jax random key for initializing network parameters
         policy_layer_sizes: sizes of hidden layers for the policy network
         critic_layer_sizes: sizes of hidden layers for the critic network
+        policy_recurrent_layer_sizes: Optionally add recurrent layers to the policy
         observation_network: optional network for processing observations.
             Defaults to utils.batch_concat.
-
     Returns:
         PPONetworks class
     """
 
     num_actions = environment_spec.actions.num_values
 
+    @hk.without_apply_rng
+    @hk.transform
     def policy_fn(inputs: jnp.ndarray) -> networks_lib.FeedForwardNetwork:
-        policy_network = hk.Sequential(
-            [
-                observation_network,
-                hk.nets.MLP(policy_layer_sizes, activation=jax.nn.relu),
-                networks_lib.CategoricalHead(
-                    num_values=num_actions, dtype=environment_spec.actions.dtype
-                ),
-            ]
+        # Add the observation network and an MLP network.
+        policy_network = [
+            observation_network,
+            hk.nets.MLP(policy_layer_sizes, activation=jax.nn.relu),
+        ]
+
+        # Add optional recurrent layers
+        if len(policy_recurrent_layer_sizes) > 0:
+            for size in policy_recurrent_layer_sizes:
+                policy_network.append(hk.GRU(size))
+
+        # Add a categorical value head.
+        policy_network.append(
+            networks_lib.CategoricalHead(
+                num_values=num_actions, dtype=environment_spec.actions.dtype
+            )
         )
-        return policy_network(inputs)
 
+        if len(policy_recurrent_layer_sizes) > 0:
+            return hk.DeepRNN(policy_network)(inputs[0], inputs[1])
+        else:
+            return hk.Sequential(policy_network)(inputs)
+
+    @hk.without_apply_rng
+    @hk.transform
+    def initial_state_fn() -> List[jnp.ndarray]:
+        state = []
+        for size in policy_recurrent_layer_sizes:
+            state.append(hk.GRU(size).initial_state(1))
```

Comment:
What are your thoughts on being able to passing something like a `recurrent_architecture_fn` (not necessarily the name to use) where we can pass in `hk.GRU` and `hk.LSTM` so that we can easily toggle between recurrent network types?
This line would then become `state.append(recurrent_architecture_fn(size)).initial_state(1)) for example.

Reply:
That sounds like a good idea:slightly_smiling_face: I updated the code.

Reply:
thank you :)